design_docs/selfhealing_featuretoggle.md

```diff
@@ -0,0 +1,160 @@
+
+
+# Prerequisites
+
+
+
+*   Running Keptn Installation
+*   Running [Unleash](https://unleash.github.io/) server as the feature toggle framework
+*   Application with feature flags connected to the Unleash server
+*   Remediation file with feature flags to toggle in case of issues
+*   Running monitoring solution for the application (right now Dynatrace is supported)
+
+
+# Remediation file
+
+The remediation file is necessary to define which remediation action should be taken in response to a problem ticket that is sent to Keptn. Keptn’s internal remediation service will then determine the actions to take and trigger them accordingly. 
+
+Here’s an example of a remediation file:
+
+remediation.yaml
+
+```yaml
+remediations:
+- name: "responsetime_p90_sockshop_carts"
+  actions:
+  - action: "featuretoggle"
+    value: "enablecache:on"
+```
+
+
+*   `name`: it is the name of the violated SLO or type of problem.
+*   `actions`: a list of actions that can be executed to remediate the problem
+*   `action`: name of the action as well as the value which should be set. I.e., the action should be set to `featuretoggle`, the value defines which feature toggle as well as the new value the feature toggle should be set to. In this case, the name of the feature toggle is `enablecache` and the value will be set to `on` once this remediation is executed.
+
+
+# User Walkthrough
+
+
+## Setup & Configure Keptn 
+
+As you would normally do with Keptn, create a project, onboard a service, and deploy your service into your Kubernetes cluster.
+
+<details><summary>Instructions</summary>
+
+### Create project 
+
+Create a Keptn project which where you want to onboard your service. Define all stages you want to have in your project in a shipyard file, similar to the [Keptn docs](https://keptn.sh/docs/0.5.0/usecases/onboard-carts-service/#create-project-sockshop).
+
+```
+keptn create project PROJECTNAME --shipyard=shipyard.yaml
+```
+
+
+
+
+
+### Onboard a service
+
+Onboard the service and add the remediation file as a resource.
+
+
+```
+keptn onboard service SERVICENAME --project=PROJECTNAME --chart=HELMCHART
+```
+
+
+Configure Keptn - add the remediation file
+
+
+```
+keptn add-resource --project=PROJECTNAME --service=SERVICENAME --stage=STAGENAME --resource=remediation.yaml
+```
+
+Deploy your application to Keptn.
+
+```
+keptn send event new-artifact --project=PROJECTNAME --service=SERVICENAME --image=docker.io/IMAGENAME --tag=TAG
+```
+
+</details>
+
+
+### Add secret for Unleash server
+
+In order to let the Keptn built-in Unleash remediation service communicate with the Unleash server, provide the credentials in terms of a Kubernetes secret.
+
+unleash-secret.yaml:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: unleash-secret
+type: Opaque
+data:
+  username: unleash-username
```

Comment:
Is there a documentation from Unleash we can link to on how to get that information?

Reply:
added some clarification